### PR TITLE
Apply the principle of least privilege to our GitHub action tokens

### PR DESCRIPTION
**Pull request by BillWagner**
I reviewed the set of GitHub actions and workflows enabled in our repo. Where possible, I lowered the permission of the GitHub token.

1. On workflows triggered by a PR, change `pull_request_target` to `pull_request` where possible. Only use`pull_request_target` on those workflows that must execute with the PR's changes. 
2. Set top level workflow permissions to `content: read`. This minimize the permissions for composite jobs.
3. Remove any unnecessary permissions on the job level permissions.

### CODE REVIEW

**Review by IEvangelist — APPROVED** [review]

**Comment by adegeo** [comment]
Which workflow updates initial comment with preview links? Is it `check-for-build-warnings`? If it is, I think it still needs actions/checkout because it's scanning for the article title, right?

**Comment by BillWagner** [comment]
> Which workflow updates initial comment with preview links? Is it `check-for-build-warnings`? If it is, I think it still needs actions/checkout because it's scanning for the article title, right?

Yes, it's check-for-build-warnings.  I think it's fine. It can still read the PR contents. Once we merge, I'll check it, and if so quickly follow by reverting that change.

File: .github/workflows/rebase-needed.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ name: "rebase required"
 
 on:
   push:
-  pull_request_target:
```
Comment (adegeo) [inline .github/workflows/rebase-needed.yml:5]: Wouldn't this have to run on the target to determine if a rebase was needed?

Comment (adegeo) [reply .github/workflows/rebase-needed.yml:5]: Agreed!

**Review by adegeo — APPROVED** [review]

**Pull request merged**